examples/ecs/nondeterministic_system_order.rs

```diff
@@ -0,0 +1,85 @@
+//! By default, Bevy systems run in parallel with each other.
+//! Unless the order is explicitly specified, their relative order is nondeterministic.
+//!
+//! In many cases, this doesn't matter and is in fact desirable!
+//! Consider two systems, one which writes to resource A, and the other which writes to resource B.
+//! By allowing their order to be arbitrary, we can evaluate them greedily, based on the data that is free.
+//! Because their data accesses are **compatible**, there is no **observable** difference created based on the order they are run.
+//!
+//! But if instead we have two systems mutating the same data, or one reading it and the other mutating,
+//! than the actual observed value will vary based on the nondeterministic order of evaluation.
+//! These observable conflicts are called **system execution order ambiguities**.
+//!
+//! This example demonstrates how you might detect and resolve (or silence) these ambiguities.
+
+use bevy::{app::AppExit, ecs::schedule::ReportExecutionOrderAmbiguities, prelude::*};
+
+fn main() {
+    App::new()
+        // This resource controls the reporting strategy for system execution order ambiguities
+        .insert_resource(ReportExecutionOrderAmbiguities)
+        .init_resource::<A>()
+        .init_resource::<B>()
+        // This pair of systems has an ambiguous order,
+        // as their data access conflicts, and there's no order between them.
+        .add_system(reads_a)
+        .add_system(writes_a)
+        // This trio of systems has conflicting data access,
+        // but it's resolved with an explicit ordering
+        .add_system(adds_one_to_b)
+        .add_system(doubles_b.after(adds_one_to_b))
```

Comment:
there seems to be only two systems in this trio

Reply:
Good call: I'll be more explicit.